mantidimaging/gui/windows/spectrum_viewer/spectrum_widget.py

```diff
@@ -38,6 +41,20 @@ def __init__(self, name: str, sensible_roi: SensibleROI, *args, **kwargs):
         self.addScaleHandle([0, 1], [1, 0])
         self._selected_row = None
 
+        menu = self.getMenu()
+        change_color_action = QAction("Change ROI Colour", self)
+        change_color_action.triggered.connect(self.onChangeColor)
+        menu.addAction(change_color_action)
+        menu.exec_(QCursor.pos())
```

Comment:
This should work, but looking into the PyQtGraph [ROI source](https://pyqtgraph.readthedocs.io/en/latest/_modules/pyqtgraph/graphicsItems/ROI.html), it seems that they have tied the logic for displaying the menu to whether the ROI is `removable` from the context menu. So it will have to be done in a more hacky way until PyQtGraph is fixed.

So in the `__init__()` replace those 5 lines with
```python
        self.menu = QtWidgets.QMenu()
        change_color_action = QAction("Change ROI Colour", self)
        change_color_action.triggered.connect(self.onChangeColor)
        self.menu.addAction(change_color_action)
```
Then we need to add a method to prevent the context menu being disabled
```python
    def contextMenuEnabled(self):
        return True
```

Reply:
(I have reported this limitation to pyqtgraph https://github.com/pyqtgraph/pyqtgraph/issues/2931 )